### PR TITLE
use portal in storage webserver

### PR DESCRIPTION
**Pull request by havardpe**
@vekterli please review

### CODE REVIEW

File: storage/src/vespa/storage/frameworkimpl/status/statuswebserver.cpp [hunk 1]
```diff
@@ -236,37 +144,39 @@ namespace {
...
                 }
             } catch (std::exception &e) {
-                HttpErrorWriter writer(out, "500 Internal Server Error");
-                writer << "<pre>" << e.what() << "</pre>";
-                pageExisted = true;
-            }
-            if (pageExisted) {
-                LOG(spam, "Status finished request");
-                return;
+                request.respond_with_error(500, "Internal Server Error");
```
Comment (vekterli) [inline storage/src/vespa/storage/frameworkimpl/status/statuswebserver.cpp]: Would be nice with a log warning here with the exception message to highlight any funny business going on internally

File: storage/src/vespa/storage/frameworkimpl/status/statuswebserver.h [hunk 2]
```diff
@@ -72,14 +76,11 @@ class StatusWebServer : private config::IFetcherCallback<vespa::config::content:
...
-    void handlePage(const framework::HttpUrlPath&, std::ostream& out);
-    static vespalib::string getServerSpec(const vespalib::string &requestSpec,
-                                          const vespalib::string &serverSpec);
+    int getListenPort() const;
+    void handlePage(const framework::HttpUrlPath&, vespalib::Portal::GetRequest request);
 private:
     void configure(std::unique_ptr<vespa::config::content::core::StorStatusConfig> config) override;
-    void getPage(const char* url, Fast_HTTPConnection& conn);
-    void run(framework::ThreadHandle&) override;
+    void run(framework::ThreadHandle&) override {}
```
Comment (vekterli) [inline storage/src/vespa/storage/frameworkimpl/status/statuswebserver.h]: It looks safe to remove the inheritance from `Runnable` entirely, but that can be done as a follow-up.

File: storage/src/vespa/storage/frameworkimpl/status/statuswebserver.cpp [hunk 3]
```diff
@@ -112,45 +72,47 @@ void StatusWebServer::configure(std::unique_ptr<vespa::config::content::core::St
...
-                    "<html><head><title>" << error << "</title></head>\r\n"
-                    "<body><h1>" << error << "</h1>\r\n"
-                    "<p>";
-        }
+struct HandleGetTask : vespalib::Executor::Task {
+    vespalib::Portal::GetRequest request;
+    std::function<void(vespalib::Portal::GetRequest)> fun;
+    HandleGetTask(vespalib::Portal::GetRequest request_in,
+                  std::function<void(vespalib::Portal::GetRequest)> fun_in)
+        : request(std::move(request_in)), fun(fun_in) {}
```
Comment (vekterli) [inline storage/src/vespa/storage/frameworkimpl/status/statuswebserver.cpp]: Since `fun_in` is passed by value, consider constructing `fun` by move

**Review by vekterli — APPROVED** [review]
:+1: only a couple of (very) minor comments

File: storage/src/vespa/storage/frameworkimpl/status/statuswebserver.h (hunk 2 above)
Comment (havardpe) [reply storage/src/vespa/storage/frameworkimpl/status/statuswebserver.h]: will do. I missed the separate inheritance when just chopping away code earlier.

File: storage/src/vespa/storage/frameworkimpl/status/statuswebserver.cpp (hunk 3 above)
Comment (havardpe) [reply storage/src/vespa/storage/frameworkimpl/status/statuswebserver.cpp]: done

File: storage/src/vespa/storage/frameworkimpl/status/statuswebserver.cpp (hunk 1 above)
Comment (havardpe) [reply storage/src/vespa/storage/frameworkimpl/status/statuswebserver.cpp]: agree

**Comment by havardpe** [comment]
@vekterli please have another look

**Review by vekterli — APPROVED** [review]
🦄👍

**Pull request merged**